File: backend/models/delivery.go
package models

// Delivery é uma estrutura que representa uma entrega no sistema.
type Delivery struct {
	ID          int     `json:"id"`          // ID único da entrega
	ClienteID   int     `json:"cliente_id"`  // ID do cliente associado à entrega
	Peso        float64 `json:"peso"`        // Peso da entrega (em kg)
	Endereco    string  `json:"endereco"`    // Endereço completo da entrega
	Logradouro  string  `json:"logradouro"`  // Nome da rua, avenida, etc.
	Numero      string  `json:"numero"`      // Número do endereço
	Bairro      string  `json:"bairro"`      // Bairro do endereço
	Complemento string  `json:"complemento"` // Complemento do endereço (ex: apartamento, bloco)
	Cidade      string  `json:"cidade"`      // Cidade do endereço
	Estado      string  `json:"estado"`      // Estado (UF) do endereço
	Pais        string  `json:"pais"`        // País do endereço
	Latitude    float64 `json:"latitude"`    // Latitude da localização da entrega
	Longitude   float64 `json:"longitude"`   // Longitude da localização da entrega
}

File: frontend/types/delivery.ts
// Tipo para o cliente
export interface Cliente {
    nome: string;
    email: string;
    telefone: string;
    cpf: string;
  }
  
  // Tipo para a entrega
  export interface Entrega {
    id: number;
    cliente_id: number;
    peso: number;
    endereco: string;
    logradouro: string;
    numero: string;
    bairro: string;
    complemento: string;
    cidade: string;
    estado: string;
    pais: string;
    latitude: number;
    longitude: number;
  }
  
  // Tipo para a requisição POST
  export interface PostDeliveryRequest {
    delivery: {
      peso: number;
      endereco: string;
      logradouro: string;
      numero: string;
      bairro: string;
      complemento: string;
      cidade: string;
      estado: string;
      pais: string;
      latitude: number;
      longitude: number;
    };
    cliente: Cliente;
  }